cmd/osbuild-service-maintenance/aws.go

```diff
@@ -112,17 +128,73 @@ func AWSCleanup(maxConcurrentRequests int, dryRun bool, accessKeyID, accessKey s
 	} else {
 		logrus.Info("Dry run, didn't actually terminate any instances")
 	}
+
 	return nil
 }
 
-func filterReservations(reservations []ec2types.Reservation) []string {
-	var instanceIDs []string
+func filterReservations(instanceIDs []string, reservations []ec2types.Reservation) []string {
 	for _, res := range reservations {
 		for _, i := range res.Instances {
 			if i.LaunchTime.Before(time.Now().Add(-time.Hour * 2)) {
-				instanceIDs = append(instanceIDs, *i.InstanceId)
+				logrus.Infof("Instance %s is too old", *i.InstanceId)
+				if !slices.Contains(instanceIDs, *i.InstanceId) {
+					instanceIDs = append(instanceIDs, *i.InstanceId)
+				}
 			}
 		}
 	}
 	return instanceIDs
 }
+
+func getChildAndParentInstances(reservations []ec2types.Reservation) []struct {
```

Comment:
I think defining this struct up to instead of an anonymous struct would make it more readable.